### DISCUSSION TITLE
Use dynamic icons with component :is

### DISCUSSION THREAD

**Discussion by dbzx10299** [nuxt/nuxt#21016] [Questions] [answered]
I have a file that should render out a code block and I pass a prop in called languages from my markdown file

```
::CodeBlock
---
languages: ['Javascript', 'Typescript']
---
::
```

Then in the `CodeBlock` component I have the following code
```js
<script setup>
const props = defineProps({
  languages: {
    type: Array,
    required: false,
    default: () => []
  }
})

const activeIndex = ref(0)

const icon = computed(() => `Icon${props.languages[activeIndex.value]}`)
</script>

<template>
  <!-- ... -->
  <component :is="icon"></component>
</template>
```

This outputs in the chrome devtools
```html
<iconjavascript></iconjavascript>
```

I have a folder called `Icon` in the components directory and a file called `Javascript.vue` and `Typescript.vue`, I wanted to dynamically update the icon based on the active index and it is updating, just that it is updating wrong. It toggles between `iconjavascript` and `icontypescript` rather than just outputting the icon.

At the moment I have to list a bunch of language icons and I feel there has to be a better way to do this.

**Comment by maou-shonen** [comment, marked as answer]
It is mentioned in this section of the documentation.
https://nuxt.com/docs/guide/directory-structure/components#dynamic-components

**Reply by dbzx10299** [reply]
It doesn't seem to be too dynamic if you can't pass a variable into the resolveComponent helper

**Reply by maou-shonen** [reply]
I remember this should work.
```ts
computed(() => resolveComponent(`Icon${props.languages[activeIndex.value]}`))
```

**Reply by dbzx10299** [reply]
That does not work, the reason is you can only pass in a normal string to resolveComponent

This will not even work with backticks
```js
resolveComponent(`IconJavaScript`)
```

Only regular quotes work
```js
resolveComponent('IconJavaScript')
```

**Reply by maou-shonen** [reply]
you are right,
Including this in the documentation would be nice.

**Reply by TheAlexLichter** [reply]
This is covered in the warning box on the linked doc page

> If you are using resolveComponent to handle dynamic components, make sure not to insert anything but the name of the component, which must be a string and not a variable.
